i3lock.c

```diff
@@ -811,7 +818,10 @@ int main(int argc, char *argv[]) {
                 unlock_indicator = false;
                 break;
             case 'i':
-                image_path = strdup(optarg);
+                if (optarg == NULL)
+                    image_path = "";
```

Comment:
The only reason this doesn't cause a crash is because we never free the string. Still not pretty, though.